components/src/core/AutoComplete/AutoComplete.tsx

```diff
@@ -0,0 +1,368 @@
+import React, { useRef, useState } from 'react';
+import {
+    TextStyle,
+    StyleProp,
+    StyleSheet,
+    View,
+    ScrollView,
+    TouchableOpacity,
+    ViewStyle,
+    TextInput as RNTextInput,
+    ViewProps,
+    TextInputProps,
+    TouchableHighlight,
+} from 'react-native';
+import { HelperText, Text } from 'react-native-paper';
+import { $DeepPartial } from '@callstack/react-theme-provider';
+import { ExtendedTheme, useExtendedTheme } from '@brightlayer-ui/react-native-themes';
+import { Chip, ChipProps } from '../Chip';
+import MatIcon from 'react-native-vector-icons/MaterialIcons';
+
+export type AutocompleteProps = ViewProps & {
+    /** Text to display the Helper Text */
+    helperText: string;
+    /** Text to display as component label */
+    label?: string;
+    /** List of Options to show in dropdown */
+    options?: string[];
+    /** Props to spread to the TextInput component. */
+    tagInputFieldProps?: TextInputProps;
+    /** Props to spread to the Chip component. */
+    chipProps?: ChipProps;
+    /** Number of Chip to be shown
+     *
+     *  @default: 6
+     *
+     */
+    limitTags?: number;
+    /** Number of character count for a Chip
+     *
+     *  @default: 16
+     *
+     */
+    limitCharacterCountTag?: number;
+    /** Callback for when the text in the Textinput changes */
+    onChange?: (details?: string[]) => void;
+    /** Callback for when the chip close icon is clicked */
+    onDelete?: (option: string) => void;
+    /** Prop to disable the AutoComplete Component
+     *
+     *  @default: false
+     *
+     */
+    disabled?: boolean;
+    /** List of  pre-populated chips to display inside TextField */
+    value?: string[];
+    /** Prop to let user pass a custom text to chip (inCase of false can only pass text from options)
+     *
+     *  @default: false
+     *
+     */
+    addCustomTag?: boolean; // if this is true only then the user can add new tags default false
```

Comment:
remove this `// if this is true only then the user can add new tags default false`